### ISSUE TITLE
Remove quantized sigmoid module

### ISSUE THREAD

**Issue by jerryzh168** [pytorch/pytorch#84290]
### 🐛 Describe the bug

sigmoid is an op that works for both fp32 and quantized input: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/native/native_functions.yaml#L4519

but we have a quantized sigomid module: https://github.com/pytorch/pytorch/blob/master/torch/ao/nn/quantized/modules/activation.py#L129-L148 this is confusing and we should remove it

## How to fix?
remove this module and make sure nothing breaks

### Versions

master

**Comment by vadimkantorov** [comment]
Also, a probable bug nearby, just below:
```python
class Softmax(torch.nn.Softmax):
    def __init__(self, dim=None, scale=1.0, zero_point=0):
        super().__init__()
        self.dim = dim
```

probably should just be 

```python
class Softmax(torch.nn.Softmax):
    def __init__(self, dim=None, scale=1.0, zero_point=0):
        super().__init__(dim = dim)
```

Also, what's situation about dim=None for nn.Softmax? In https://github.com/pytorch/pytorch/issues/36524 is discussed that implicit dim for F.softmax is deprecated. What's the behavior for dim=None? Is then softmax computed across all of tensor?

It seems in https://pytorch.org/docs/stable/generated/torch.nn.Softmax.html?highlight=nn%20softmax#torch.nn.Softmax that dim=None is supported (specified as default value), but further down, dim is specified as int, and None isn't documented as supported value.

IMO dim=None should be deprecated to match F.softmax

**Comment by jerryzh168** [comment]
Hi @vadimkantorov can you open a new issue for problems related to nn.Softmax?

**Comment by jerryzh168** [comment]
actually I think we should remove Softmax as well

**Comment by vadimkantorov** [comment]
> Hi @vadimkantorov thanks for pointing out the problem of quantized softmax implementation, can you open a new issue for problems related to floating point nn.Softmax?

done

**Comment by vadimkantorov** [comment]
About this PR in general: Softmax / Sigmoid ops change the values range significantly (by clamping it between 0 and 1 and for softmax it's typically even more severe to [0, 0.1 or 0.01]. Currently these modules allow for a different zero_point / scale for output quantized tensor, will this change if the nnq modules are removed

**Comment by andrewor14** [comment]
@jerryzh168 Looks like the lines in your PR description have shifted. Do you mean these lines? https://github.com/pytorch/pytorch/blob/c435b8c10a1b9652a35e1a87cee6c3942356660d/torch/ao/quantization/quantization_mappings.py#L192-L197
